src/dawn/Optimizer/PassMultiStageSplitter.cpp

```diff
@@ -33,103 +33,157 @@ bool PassMultiStageSplitter::run(
     const std::shared_ptr<StencilInstantiation>& stencilInstantiation) {
   OptimizerContext* context = stencilInstantiation->getOptimizerContext();
 
-  int numSplit = 0;
-  std::deque<MultiStage::SplitIndex> splitterIndices;
-  DependencyGraphAccesses graph(stencilInstantiation.get());
-
-  // Iterate over all stages in all multistages of all stencils
-  for(auto& stencil : stencilInstantiation->getStencils()) {
-
-    int multiStageIndex = 0;
-    for(auto multiStageIt = stencil->getMultiStages().begin();
-        multiStageIt != stencil->getMultiStages().end(); ++multiStageIndex) {
-      MultiStage& multiStage = (**multiStageIt);
-
-      splitterIndices.clear();
-      graph.clear();
-
-      // Loop order specified by the user in the vertical region (e.g {k_start, k_end} is forward)
-      auto userSpecifiedLoopOrder = multiStage.getLoopOrder();
+  if(!context->getOptions().Debug) {
```

Comment:
this very large if/elses make the code a bit unreadable. Can we dispatch the code to private methods: ?
```
if(!context->getOptions().Debug) {
   solveMultiStageSplit()
else 
   solveMultiStageSplitDebug();
}
```